craft_parts/errors.py

```diff
@@ -688,3 +693,19 @@ def __init__(self, warning_list: Iterable[str]) -> None:
             ),
         )
         Warning.__init__(self)
+
+
+class PartitionNotFound(PartitionUsageError):
+    """A partition has been specified that does not exist.
+
+    :param partition_name: The name of the partition that does not exist.
+    :param partitions: Iterable of the names of valid partitions.
+    """
+
+    def __init__(self, partition_name: str, partitions: Iterable[str]) -> None:
+        self.partition_name = partition_name
```

Comment:
Do you need this line?

Reply:
Yep I do, it's for callers.  Added a comment clarifying.